src/server/transaction.h

```diff
@@ -499,25 +513,10 @@ class Transaction {
   // synchronize the multi-shard transaction.
   uint32_t CalcMultiNumOfShardJournals() const;
 
-  void WaitForShardCallbacks() {
-    run_ec_.await([this] { return 0 == run_count_.load(std::memory_order_relaxed); });
-
-    // no reads after this fence will be reordered before it, and if a store operation sequenced
-    // before some release operation that happened before the fence in another thread, this store
-    // will be visible after the fence.
-    // In this specific case we synchronize with DecreaseRunCnt that releases run_count_.
-    // See #997 before changing it.
-    std::atomic_thread_fence(std::memory_order_acquire);
```

Comment:
Isn't 
```c++
atomic.load(relaxed)
fence(acquire)
```

the same as just?
```c++
atomic.load(acquire)
```

I assume that using fences directly makes only sense when optimizing batch operations around atomics (i.e. lot's of operations before a single message passing)

Or does it make a difference because we run `load()` withing a ec_ more than once? (almost always twice)🤓 I suppose it doesn't have any impact because it's mostly a compiler fence here and we don't have many reads around

Reply:
atomic_thread_fence is not a compiler fence

Reply:
Every atomic fence is a compiler fence, on x86 if often actually nothing more than than 🙂

> On x86 (including x86-64), atomic_thread_fence functions issue no CPU instructions and only affect compile-time code motion, except for std::atomic_thread_fence(std::memory_order::seq_cst)

https://en.cppreference.com/w/cpp/atomic/atomic_thread_fence

Reply:
what you said, I wanted to avoid acquire load inside event count callback. probably over-optimization